sklearn/base.py

```diff
@@ -249,6 +254,67 @@ def get_params(self, deep=True):
             out[key] = value
         return out
 
+    def _get_params_html(self, deep=True):
+        """
+        Get parameters for this estimator with a specific HTML representation.
+
+        Parameters
+        ----------
+        deep : bool, default=True
+            If True, will return the parameters for this estimator and
+            contained subobjects that are estimators.
+
+        Returns
+        -------
+        params : ParamsDict
+            Parameter names mapped to their values. We return a `ParamsDict`
+            dictionary, which renders a specific HTML representation in table
+            form.
+        """
+        out = self.get_params(deep=deep)
+
+        init_func = getattr(self.__init__, "deprecated_original", self.__init__)
+        init_default_params = inspect.signature(init_func).parameters
+        init_default_params = {
+            name: param.default for name, param in init_default_params.items()
+        }
+
+        def is_non_default(param_name, param_value):
+            """Finds the parameters that have been set by the user"""
+            if (
+                param_name not in init_default_params
+            ):  # happens if k is part of a **kwargs
+                return True
+            if (
+                init_default_params[param_name] == inspect._empty
+            ):  # k has no default value
+                return True
+            # try to avoid calling repr on nested estimators
+            if (
+                isinstance(param_value, BaseEstimator)
+                and param_value.__class__ != init_default_params[param_name].__class__
+            ):
+                return True
+
+            if param_value != init_default_params[param_name] and not (
+                is_scalar_nan(init_default_params[param_name])
+                and is_scalar_nan(param_value)
+            ):
+                return True
+            return False
+
+        # reorder the parameters from `self.get_params` using the `__init__`
+        # signature
+        remaining_params = [name for name in out if name not in init_default_params]
+        ordered_out = {name: out[name] for name in init_default_params if name in out}
+        ordered_out.update({name: out[name] for name in remaining_params})
+
+        non_default_ls = [
+            name for name, value in ordered_out.items() if is_non_default(name, value)
+        ]
```

Comment:
Let's keep a tuple since by default `non_default` in `ParamsDict` is a tuple.

```suggestion
        non_default_params = tuple([
            name for name, value in ordered_out.items() if is_non_default(name, value)
        ])
```